internal/handlers/common/filter.go

```diff
@@ -55,6 +55,11 @@ func FilterDocument(doc, filter *types.Document) (bool, error) {
 
 // filterDocumentPair handles a single filter element key/value pair {filterKey: filterValue}.
 func filterDocumentPair(doc *types.Document, filterKey string, filterValue any) (bool, error) {
+	if strings.HasPrefix(filterKey, "$") {
```

Comment:
This check is moved to the top of this function to handle cases like `path.$something`. Without this change, `-disable-pushdown=false` and `-disable-pushdown=true` would give different results (cc @chilagrow, it's relevant to your task a bit).

Reply:
Thanks!

Reply:
That is actually quite concerning: `path.$something` should return the same result with pushdown enabled or disabled – an error about invalid path / field name. And I think we should a missing check for in the right place and add tests, not just move `if`

Reply:
Interesting... I moved this back, and I can't find this case anymore. So, maybe it was something else. 

Then I don't introduce this change, but I added a couple of more tests with `$`.